drivers/include/periph/doc.txt

```diff
@@ -16,6 +16,64 @@
  * defines a standardized interface to access MCU peripherals that is not tied
  * to any specific vendor, platform or architecture.
  *
+ * # Initialization and Thread Safety
+ *
+ * Peripheral drivers fall into three categories regarding to their
+ * initialization:
+ *
+ * 1. Drivers that do not need to be initialized
+ *    (e.g. @ref drivers_periph_eeprom)
+ * 2. Drivers that are automatically initialized by the module `periph_init`,
+ *    (see @ref drivers_periph_init for details)
```

Comment:
```suggestion
 * 2. Drivers that are automatically initialized by the module `periph_init` at boot,
 *    (see @ref drivers_periph_init for details)
```

In the paragraph below it mentions drivers initialized at boot. Better make it explicit which category this is.